genderize.py

```diff
@@ -0,0 +1,184 @@
+"""
+Created on Tue Sep 10 14:07:15 2019
+@author: tickc
+"""
+
+
+import csv
+import os 
+import nltk
+import random
+#############################################################################
+#
+# A necessary utility for accessing the data local to the installation.
+#
+#############################################################################
+
+_ROOT = os.path.abspath(os.path.dirname(__file__))
+def get_data(path):
+    return os.path.join(_ROOT, 'data', path)
+get_data("filename")
+
+#using .split() will give incorrect values since it does not split whitespace
+def split(values): 
+    return [char for char in values] 
+
+def remove_vowels(s):
+    vowels = "aeiouAEIOU"
+    s_without_vowels = ""
+    for letter in s:
+        if letter not in vowels:
+            s_without_vowels += letter
+    return s_without_vowels
+
+def only_vowels(s):
+    vowels = "aeiouAEIOU"
+    only_vowels = ""
+    for letter in s:
+        if letter in vowels:
+            only_vowels += letter
+    return only_vowels
+
+def extract_features(name):
+    
+    name = name.lower()
+    return {
+        'last_char': name[-1],
+        'last_two': name[-2:],
+        'last_three': name[-3:],
+        'first': name[0],
+        'first2': name[:2],
+        'first3': name[:3],
+        'no_vowels': remove_vowels(name),
+        'only_vowels': only_vowels(name),
+    }
+
+
+    
+class genderizer:
+
+    def __init__(self):
+        with open(get_data("country_pop.csv")) as g:
+            self.pop = dict(filter(None, csv.reader(g)))
+        
+        self.f = open(get_data("nam_dict.txt"))
```

Comment:
Temporary file handles used to read data into class variables should not be member variables themselves--you don't need self.f past the end of the init function, so just call it f and use it as normal.